uint8 dist1
uint8 dist2
uint8 dist3
uint8 dist4
uint8 dist5
uint8 dist6
